tests/e2e/download/download.spec.js

```diff
@@ -0,0 +1,126 @@
+import { test, expect } from 'playwright-test-coverage'
+
+import collectionsGraphJson from './__mocks__/collections_graph.json'
+import timeline from './__mocks__/timeline.json'
+import granules from './__mocks__/granules.json'
+import retrievals from './__mocks__/retrievals.json'
+import retrieval from './__mocks__/retrieval.json'
+
+import { login } from '../../support/login'
+import { getAuthHeaders } from '../../support/getAuthHeaders'
+
+test.describe('Download spec', () => {
+  test('get to the download page', async ({ page, context }) => {
+    login(context)
+
+    const authHeaders = getAuthHeaders()
+
+    await page.route(/graphql/, async (route) => {
+      await route.fulfill({
+        json: collectionsGraphJson.body,
+        headers: authHeaders
+      })
+    })
+
+    await page.route(/timeline/, async (route) => {
+      await route.fulfill({
+        json: timeline.body,
+        headers: authHeaders
+      })
+    })
+
+    // Load the download data and mock the results
+    await page.route(/retrievals/, async (route) => {
+      if (route.request().method() === 'GET') {
+        await route.fulfill({
+          json: retrieval.body,
+          headers: retrieval.headers
+        })
+      } else {
+        await route.fulfill({
+          json: retrievals.body,
+          headers: retrievals.headers
+        })
+      }
+    })
+
+    await page.route(/granule_links/, async (route) => {
```

Comment:
```suggestion
       await page.route(/granule_links/, async (route) => {
      await route.fulfill({
        json: {
          cursor: 'mock-cursor',
          links: {
            browse: [],
            download: [
              'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/006/MYD04_3K.A2020006.1720.061.2020008170450.hdf',
              'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/006/MYD04_3K.A2020006.1900.061.2020008170003.hdf',
              'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/007/MYD04_3K.A2020007.1805.061.2020008182434.hdf',
              'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/008/MYD04_3K.A2020008.1850.061.2020010183913.hdf',
              'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/009/MYD04_3K.A2020009.1755.061.2020010200250.hdf',
              'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/010/MYD04_3K.A2020010.1835.061.2020011153413.hdf',
              'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/011/MYD04_3K.A2020011.1740.061.2020012150910.hdf',
              'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD04_3K/2020/012/MYD04_3K.A2020012.1825.061.2020013152621.hdf'
            ],
            s3: []
          }
        },
        headers: authHeaders
      })
    })
```
I removed the second call to match the cypress one and the test was still passing locally

Reply:
It did get to the other call though so I'm fine either way if anything I think yours is a bit more verbose and has paging more correctly mocked

Reply:
I think I like it the way that I wrote things.